packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -281,7 +276,10 @@ function unmount(vnode: VNode, parent: ParentNode, doRemove: boolean = false) {
     // When unmounting a VNode subtree not all the elements have to removed from the DOM. The
     // subtree root, is the only element worth unmounting from the subtree.
     if (doRemove) {
-        removeNode(elm!, parent);
+        // The vnode might or might not have a data.renderer associated to it
+        // but the removal used here is from the owner instead.
+        const renderer = getRendererFromVM(vnode.owner);
```

Comment:
@ravijayaramappa this is the only piece that I believe can be improve... getting the renderer from the vnode instead, but we will need to inspect the vnode.data.

Reply:
@caridy for comment and text, we do dom operations using the owner's renderer, so that case is fine. For Element and CustomElement, the vnode's renderer is used. To keep it symmetrical, should the same renderer be used.
For example: Would LWS have issues if the node is removed without its knowledge?